### ISSUE TITLE
Printglossary fix breaks build scripts like latexmk

### ISSUE THREAD

**Issue by stefanct** [egraff/ult-base#9]
Hi,
I was using your hack to keep `\printglossary` from inserting empty pages. Unfortunately there is a problem with creating the file before `\printglossary` (https://github.com/egraff/ult-base/blob/f771e58356d05aa56413c940440b62676d25328c/texmf-tds/tex/latex/ult-base/ult-printglossary.sty#L58) is called resulting in the glossary (or list of abbreviation in my case) to not be printed at all because the `.acr` file is never created correctly.

I am using [latexmk](https://ctan.org/pkg/latexmk) to automate building. It relies on parsing the stdout prints to derive the which files do not exist and launches the respective build steps (like executing `makeglossaries`). This does not longer happen which your hack. I tried to fix that by pretending that the file is not really there (by adding `\typeout{No file ##1.}`. Unfortunately this does not fool latexmk :(

Also, `\immediate\write\ult@tmp@glossary@file{}` adds a newline character to the resulting file. You can simply leave that out to create an empty file (dunno if that could a regression or fixes other use cases - JFYI). I don't have tried further to find a fix because I have no time ATM (and lost lots of it already due to this :/). Maybe patching `\@print@glossary` directly instead would work better?

BTW: why is Nicola doing that `\null` there anyway? Maybe we should ask her (she is very responsive on stackexchange).

**Comment by egraff** [comment]
This is a very interesting problem! Originally, the reason I went for the current workaround is that patching could more easily break when the package gets updated, which is harder to maintain (especially if there is a need to support older versions/installs of the package). I could have another look, though 🙂 however, I do like your suggestion about trying to ask Nicola - would you be up for writing a question on TeX stackexchange, if that is the easiest way forwards?

My main concern, if I were to change the workaround to patch the glossary command, is how to ensure the code does not regress, if the goal is to better support latexmk. I don't currently have any 'test infrastructure' that would allow me to write a test that ensures a document will build using a custom build script (and I do unfortunately not have a lot of time to spend on this). In case I do get some spare cycles, though, do you have a minimal latexmk file I could experiment with?

**Comment by stefanct** [comment]
Hi,
for testing you can use the following files after removing the `.txt` suffix. Compile with `latexmk -pdf` and clean up between commenting in/out the `\usepackage{ult-printglossary}` statement with `latexmk -C`
[latexmkrc.txt](https://github.com/egraff/ult-base/files/5122263/latexmkrc.txt)
[test.tex.txt](https://github.com/egraff/ult-base/files/5122264/test.tex.txt)

When ult-printglossary is not loading the second page shows the list of abbreviations. When it _is_ active only the title is shown on the second page. If you don't see a proper way to fix this I can post it on SA but feel free to take my example and ask yourself. You probably know better what's going on - after looking into it for a short time to formulate this report I forgot most of what I learned about it already :)

**Comment by stefanct** [comment]
Just for reference (because I would regularly run into this once in a while ;)
My SA question from last year (with a response by Ulrike Fischer): https://tex.stackexchange.com/questions/526238/why-is-glossaries-so-eager-to-insert-null

And my report on Nicola's tracker: [146](https://www.dickimaw-books.com/bugtracker.php?key=146) (which she deemed a duplicate of [102](https://www.dickimaw-books.com/bugtracker.php?key=102).

Basically this means WONTFIX. Again, I have no time to look into a feasible workaround right now. :|

**Comment by jorag** [comment]
I think I have the same problem when using the uit-thesis template in Overleaf. The glossary is not printed at least.

Did anyone find a workaround?

**Comment by jorag** [comment]
> I think I have the same problem when using the uit-thesis template in Overleaf. The glossary is not printed at least.
> 
> Did anyone find a workaround?

I got it working. I assume I have caused the issue myself by messing around with the files trying to fix something else.

What I did was to get the package from a version I knew was working: https://github.com/egraff/uit-thesis/tree/338297d50a63e6d879e85003f6e73856da0c895c

Then I uploaded the packages as folders in the Overleaf project and used the following commands in the preamble:

```
\usepackage{uit-base/ult-glossaries}
\usepackage{uit-base/ult-printglossary}
\newglossaryentry{latex}
{
    name=latex,
    description={Is a markup language specially suited 
    for scientific documents}
}
\makeglossaries
```
Then in the frontmatter:

```
\tableofcontents
\printglossary
```
